### ISSUE TITLE
dom.webcomponents.shadowdom.enabled doesn't exist

### ISSUE THREAD

**Issue by vertigo220** [meetDeveloper/Dictionary-Anywhere#40]
Addons page says to set dom.webcomponents.shadowdom.enabled to enabled, but my browser (Waterfox 2020.12 64-bit) doesn't have that; it only has dom.webcomponents.enabled, which did make a difference when setting to enabled though it still doesn't work (just says "Searching Please Wait...") as mentioned in issue #34.

**Comment by meetDeveloper** [comment]
You also will need to enable dom.getRootNode.enabled. :) Then it will work for you.

I did not knew about waterfox browser, I will add this too in addons page and in github also so that future users of this extension does not face problem when using waterfox browser. :)

**Comment by vertigo220** [comment]
Thanks! Works as expected now.

I'd also recommend adding a bit on what these two settings do and why they need to be changed. I've used lots of add-ons and don't recall ever having to change anything in about:config for them to work, including when using ones that generate pop-ups like this one does, so I wonder if there's a better way to achieve whatever is needed that requires them to be changed and I'd like to know what's happening when I change them so I know I'm not reducing security or functionality in some way.

**Comment by meetDeveloper** [comment]
I use shadow dom for the popup. I use Shadow DOM due to level of encapsulation it provides.
You can read more about this here. Shadow DOM is recently been made available in browser that is why older version of browsers does not have it and some have it behind a flag. Firefox v63.0+ have this automatically available and there is no need for any config change, not sure why Waterfox does not have it enabled by default even in its latest version.

**Comment by vertigo220** [comment]
Thanks. A few follow-up questions:

- What do you mean by encapsulation?
- Why do you use it when other add-ons achieve pop-ups without it?
- Was there supposed to be a link in your reply? ("You can read more about this here.")
- What about dom.getRootNode.enabled?

**Comment by meetDeveloper** [comment]
Yeah missed the link: https://glazkov.com/2011/01/14/what-the-heck-is-shadow-dom/

By encapsulation I mean by popup is separated from other things in page and their CSS does not affect my popup.
I use function _getRootNode_, you can read about Shadow DOM, this is regarding that only, I downloaded _WaterFox_ and tried the addon, and error was coming _getRootNode_ is not a function then I saw we have to enable this config option.

Other addon if they do achieve they are simply not using shadow dom and it may cause style from outside bleed into their popup.

**Comment by vertigo220** [comment]
Thanks again. Looks pretty neat. Didn't understand it all, but the gist seems to be that essentially it's like an island for code in the page, like an iframe for a single object.

So getRootNode must also be enabled by default in newer/other browsers, and just disabled by default in Waterfox for some reason, so calling the function failed. Interesting. Anyways, it's good to know more about these entries. Thanks for all the info. :)
